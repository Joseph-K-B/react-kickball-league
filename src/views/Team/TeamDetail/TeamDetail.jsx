import { useEffect, useState } from "react";
import { Link, useParams } from "react-router-dom";
import { getTeamById } from "../../../services/teams";

function TeamDetail() {
    const { teamId } = useParams()
    const[team, setTeam] = useState(null)

    useEffect(() => {
        getTeamById(teamId).then((res) => setTeam(res))
    }, [teamId])

    if(!team) return <h2>Loading team...</h2>

    return(
        <>
        <h1>{team.name}</h1>
        <h3>{team.city}, {team.state}</h3>
        <p>Wins: {team.wins}</p>
        <ul>
        {team.players.map((player) => (
            <li key={player.id}>
                <Link className='link' to={`/players/${player.id}`}>
                    <p>{player.position} - {player.name}</p>
                </Link>
            </li>
        ))}
        </ul>
        <Link className='link' to='/teams'><p>Team Directory</p></Link>
        </>
    )
}

export default TeamDetail