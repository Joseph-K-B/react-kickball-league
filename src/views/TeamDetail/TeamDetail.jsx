import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { getPlayers } from "../../services/players";
import { getTeamById } from "../../services/teams";

function TeamDetail() {
    const { id } = useParams()
    const[team, setTeam] = useState(null)

    useEffect(() => {
        getTeamById(id).then((res) => setTeam(res))
    }, [id])

    if(!team) return <h2>Loading team...</h2>

    return(
        <>
        <h1>{team.name}</h1>
        <h3>{team.city}, {team.state}</h3>
        <ul>
        {team.players.map((player) => (
            <li key={player.id}>{player.position} - {player.name}</li>
        ))}
        </ul>

        </>
    )
}

export default TeamDetail