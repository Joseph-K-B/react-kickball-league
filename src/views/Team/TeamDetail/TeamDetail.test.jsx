import { render, screen } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import TeamDetail from './TeamDetail'

it('renders team details according to id param', async () => {
    const {container} = render(
        <MemoryRouter initialEntries={['/teams/1']}>
            <Route path="/teams/:teamId">
                <TeamDetail />
            </Route>
        </MemoryRouter>
    );

    screen.getByText('Loading team...')

    const team = await screen.findByText(/Bridge City Sinners/, {exact: false});
    const location = await screen.findByText(/Portland, OR/, {exact: false});
    expect(team).toBeInTheDocument();
    expect(location).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})