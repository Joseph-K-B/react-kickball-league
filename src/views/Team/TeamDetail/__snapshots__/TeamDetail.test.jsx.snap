// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders team details according to id param 1`] = `
<div>
  <h1>
    Bridge City Sinners
  </h1>
  <h3>
    Portland
    , 
    OR
  </h3>
  <p>
    Wins: 
    0
  </p>
  <ul>
    <li>
      <a
        class="link"
        href="/players/4"
      >
        <p>
          Outfield
           - 
          Dangalf the Wise
        </p>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/players/2"
      >
        <p>
          Catcher
           - 
          Will I Aint
        </p>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/players/15"
      >
        <p>
          3rd Base
           - 
          Nikki Kicks
        </p>
      </a>
    </li>
  </ul>
  <a
    class="link"
    href="/teams"
  >
    <p>
      Team Directory
    </p>
  </a>
</div>
`;
